src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala

```diff
@@ -48,46 +48,30 @@ object GoServiceMainHandlersGenerator {
   private[main] def generateHandlerDecl(root: ServiceRoot, operation: CRUD): String =
     s"func (env *env) ${operation.toString.toLowerCase}${root.name}Handler(w http.ResponseWriter, r *http.Request)"
 
-  /** Generate a errMsg declaration and http.Error call */
-  private[main] def generateHTTPError(statusCodeEnum: GoHTTPStatus, errMsg: String, errMsgArgs: String*): String = {
-    val createErrorJSONargs =
+  /** Generate a respond with error call */
+  private[main] def generateRespondWithError(
+    statusCode: String,
+    metricSuffix: Option[String],
+  )(errMsg: String, errMsgArgs: String*): String = {
```

Comment:
I'm struggling to see why splitting this into two sets of arguments is useful? Why aren't all the arguments taken together?

Reply:
You're right, I originally didn't have MetricSuffix as an optional, just as a string, and was overloading by omitting it.